netrd/utilities/graph.py

```diff
@@ -72,3 +72,36 @@ def ensure_undirected(G):
         G = G.to_undirected(as_view=False)
         warnings.warn("Coercing directed graph to undirected.", RuntimeWarning)
     return G
+
+
+def ensure_unweighted(G):
+    """Ensure the graph G is unweighted.
+
+    If it is not, coerce it to unweighted and warn the user.
+
+    Parameters
+    ----------
+    G (networkx graph)
+        The graph to be checked
+
+    Returns
+    -------
+
+    G (nx.Graph)
+        Unweighted version of the input graph
+
+    """
+    H = G.copy()
+    is_weighted = False
+    for _, _, attr in H.edges(data=True):
+        if attr.get('weight', 1.0) != 1.0:
```

Comment:
Don't directly compare to `1.0`. Instead, check that the difference between the weight and `1.0` is smaller than an epsilon.

Also, the second loop is unnecessary. Just pop the weights inside this loop, whenever a weight is found.